### ISSUE TITLE
Allow preventing Modal close on background click

### ISSUE THREAD

**Issue by JacobHaitsma** [Faithlife/styled-ui#500]
There are some cases in which we may want greater control over how a modal can be closed, and adding a Modal prop to prevent 'close on background click' behavior would help achieve this.

For example, when an action is being performed that causes a user to wait (eg. adding check-ins in ChMS), preventing close on background click will cut down on accidental exits, helping ensure the user sees indication of a successful check in and any other post-checkin information within the modal by reducing the accidental exit area significantly.

(this is low priority, but if all looks well I'd be happy to implement this - just wanted to get an OK/any thoughts before starting)
